packages/core/ai_content_platform/vimax/adapters/image_adapter.py

```
@@ -0,0 +1,306 @@
+            "16:9": "landscape_16_9",
+            "9:16": "portrait_16_9",
+            "4:3": "landscape_4_3",
+            "3:4": "portrait_4_3",
+        }
+        return sizes.get(aspect_ratio, "square")
+
+    def _download_image(self, url: str, path: str):
+        """Download image from URL to path."""
+        try:
+            import urllib.request
+            urllib.request.urlretrieve(url, path)
```

Ruff (0.14.14)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)